### PR TITLE
fix: optional eds icon in viewselectoritem

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
View selector will display the eds icon in the viewselector item instead of the one in viewconfig

## Why is this pull request needed?
To have the possiblity to both control the displayed icon in the sidebar and to not change view when clicking the sidebar

### CODE REVIEW

File: example/app/data/DemoDataSource/recipes/plugins/view_selector/car_infosys/owner.recipe.json [hunk 1]
```diff
@@ -41,8 +42,7 @@
                 "type": "PLUGINS:dm-core-plugins/form/FormInput",
                 "fields": ["earlierOwners"]
               }
-            },
-            "eds_icon": "group"
+            }
```
Comment (soofstad) [inline example/app/data/DemoDataSource/recipes/plugins/view_selector/car_infosys/owner.recipe.json:45]: Should leave one of these icons in the ViewConfig, to make sure the backwards compatibility works

File: packages/dm-core-plugins/src/view_selector/Sidebar.tsx [hunk 2]
```diff
@@ -12,24 +12,20 @@ export const Sidebar = (props: {
...
             <div key={viewItem.viewId}>
               {viewItem.subItems ? (
                 <SideBar.Accordion
                   key={viewItem.viewId}
                   icon={
-                    viewItem.viewConfig?.eds_icon
-                      ? EdsIcons[
-                          viewItem.viewConfig.eds_icon as keyof typeof EdsIcons
-                        ]
+                    viewItem.eds_icon
```
Comment (soofstad) [inline packages/dm-core-plugins/src/view_selector/Sidebar.tsx:27]: This does not keep backwards compatibility

**Review by soofstad — CHANGES_REQUESTED** [review]
Looks like none of the changes keeps backwards compatibility

File: packages/dm-core-plugins/src/view_selector/Sidebar.tsx (hunk 2 above)
Comment (collinlokken) [reply packages/dm-core-plugins/src/view_selector/Sidebar.tsx:36]: Fixed

File: example/app/data/DemoDataSource/recipes/plugins/view_selector/car_infosys/owner.recipe.json (hunk 1 above)
Comment (collinlokken) [reply example/app/data/DemoDataSource/recipes/plugins/view_selector/car_infosys/owner.recipe.json]: Fixed

File: packages/dm-core-plugins/src/view_selector/Sidebar.tsx [hunk 3]
```diff
@@ -12,25 +12,35 @@ export const Sidebar = (props: {
...
     props
-
   return (
     <SideBar open style={{ height: 'auto' }}>
       <SideBar.Content>
         {viewSelectorItems.map((viewItem: TItemData) => {
           // subItem's will be rendered inside other items. Don't add them here
           if (viewItem.isSubItem) return null
-
+          console.log(
```
Comment (soofstad) [inline packages/dm-core-plugins/src/view_selector/Sidebar.tsx:21]: remove?

File: packages/dm-core-plugins/src/view_selector/Tabs.tsx [hunk 4]
```diff
@@ -20,6 +20,11 @@ export const Tabs = (props: {
     >
       <EdsTabs.List>
         {viewSelectorItems.map((config: TItemData) => {
+          console.log(
+            config.eds_icon ??
+              config.viewConfig?.eds_icon ??
+              subdirectory_arrow_right
+          )
```
Comment (soofstad) [inline packages/dm-core-plugins/src/view_selector/Tabs.tsx:27]:
```suggestion
```

**Review by soofstad — APPROVED** [review]

**Pull request merged**